rmw_connextdds_common/src/common/rmw_context.cpp

```diff
@@ -801,6 +847,22 @@ rmw_api_connextdds_init(
 
   ctx->qos_library = qos_library;
 
+  if (ctx->qos_library != "") {
+    struct DDS_StringSeq library_names = DDS_SEQUENCE_INITIALIZER;
+    if (
+      DDS_RETCODE_OK != DDS_DomainParticipantFactory_get_qos_profile_libraries(
+        ctx->factory, &library_names))
+    {
+      RMW_SET_ERROR_MSG("failed to get qos profile libraries");
+      return RMW_RET_ERROR;
+    }
+    if (!rmw_connextdds_find_string_in_list(&library_names, qos_library)) {
+      RMW_CONNEXT_LOG_ERROR_A_SET(
+        "specified qos library {%s} was not found", qos_library);
+      return RMW_RET_ERROR;
+    }
+  }
+
```

Comment:
Do we still need to load a QoS library name? AFAICT it's not used anymore, and if you can I would just get rid of `ctx->qos_library`, the code looking up its value from the environment, and the environment variable itself altogether.

While we are doing some clean-up, would you mind also deleting the internal functions to "guess profile names" in `rmw_impl.cpp`? From `#define RMW_CONNEXT_QOS_TAG_NODE ...` to function `rmw_connextdds_list_service_qos_profiles()`. They are not needed, and with this PR, we have a good way to customize QoS further.

Thanks!

Reply:
> Do we still need to load a QoS library name? AFAICT it's not used anymore, and if you can I would just get rid of ctx->qos_library, the code looking up its value from the environment, and the environment variable itself altogether.

Yes, that's true. Removed it in https://github.com/rticommunity/rmw_connextdds/pull/7/commits/22a9847053c24a268850214d787712cc78c06df3.

> While we are doing some clean-up, would you mind also deleting the internal functions to "guess profile names" in rmw_impl.cpp? From #define RMW_CONNEXT_QOS_TAG_NODE ... to function rmw_connextdds_list_service_qos_profiles(). They are not needed, and with this PR, we have a good way to customize QoS further.

Removed all that as well.